project-base/storefront/hooks/productLists/comparison/useComparison.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+    ProductListTypeEnumApi,
+    useAddProductToComparisonMutationApi,
+    useCleanProductListMutationApi,
+    useComparisonQueryApi,
+    useRemoveProductFromComparisonMutationApi,
+} from 'graphql/generated';
+import { showErrorMessage, showSuccessMessage } from 'helpers/toasts';
+import { useIsUserLoggedIn } from 'hooks/auth/useIsUserLoggedIn';
+import { useProductList } from 'hooks/productLists/useProductList';
+import useTranslation from 'next-translate/useTranslation';
+import { useEffect, useState } from 'react';
+import { usePersistStore } from 'store/usePersistStore';
+
+export const useComparison = () => {
+    const { t } = useTranslation();
+    const isUserLoggedIn = useIsUserLoggedIn();
+
+    const [, addProductToListMutation] = useAddProductToComparisonMutationApi();
+    const [, removeProductFromListMutation] = useRemoveProductFromComparisonMutationApi();
+    const [, cleanListMutation] = useCleanProductListMutationApi();
+
+    const comparisonUuid = usePersistStore((store) => store.comparisonUuid);
```

Comment:
That's pretty big question, maybe I am missing some understanding. Seems like we are complicating things on FE. Since on BE we have only one API and on FE we are basically splitting it into two APIs/logics back again. 

Why don't we have type next to uuid coming from API and then we just filter out proper type and use its uuid? Then we can have only one value in store like `productListUuids` and store uuids with proper type. I guess it's connected with having two different fragments/structures, but seems to me like huge improvement in case we share with BE only one API because now we need to consider the type almost everywhere... So to me it seems like we have done half of the possible improvements as was done on BE.

Reply:
As discussed, I agree with you. After going through it once again, I moved the layer of generic code up, now making basically everything generic. You can check [the commit, in which I do it](https://github.com/shopsys/shopsys/pull/2901/commits/40f0a750f24c4d5c83d4bfc417cf203a2727053d) and the modified upgrade notes.